packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -420,81 +445,75 @@ class _DecoyChild extends StatefulWidget {
 }
 
 class _DecoyChildState extends State<_DecoyChild> with TickerProviderStateMixin {
-  // TODO(justinmc): Dark mode support.
-  // See https://github.com/flutter/flutter/issues/43211.
-  static const Color _lightModeMaskColor = Color(0xFF888888);
-  static const Color _masklessColor = Color(0xFFFFFFFF);
-
-  final GlobalKey _childGlobalKey = GlobalKey();
-  late Animation<Color> _mask;
   late Animation<Rect?> _rect;
+  late Animation<Decoration> _boxDecoration;
+
+  BorderRadiusGeometry getBorderRadius() {
+    switch(widget.child.runtimeType) {
+      case Container:
+        return ((widget.child as Container?)?.decoration as BoxDecoration?)?.borderRadius ?? BorderRadius.circular(0);
+      case DecoratedBox:
+        return ((widget.child as DecoratedBox?)?.decoration as BoxDecoration?)?.borderRadius ?? BorderRadius.circular(0);
+      default:
+        return BorderRadius.circular(0);
+    }
+  }
```

Comment:
I'm running out of ideas on how to best get the border radius. It's necessary in order to have it not look weird with the decoy shadow lining up with the child. This way works, but I think also needs a check for other kinds of Decoration types in case those are used, and maybe needs a `try` block, but this feels incorrect. I was hoping to find a way to see if an instance of a parent class (i.e. `Widget`) has a getter belonging to some but not all of it's possible subclasses that it could also be (i.e. `Container` or `DecoratedBox`), but it seems like Dart doesn't do that?

I guess I could get rid of the shadow effect for now, and leave it not matching native 100%, but I'd like to avoid that.

Reply:
Here's how I think this should work:

By default, handle the simple case where the child has no opinion about its border radius or shadow.  So your plain red square example should work nicely out of the box and animate to having a border radius when opened, with no type checking or anything smart needed.

For anything more complicated than that, allow the user to control the animation themselves.  This is what I was trying to achieve with [previewBuilder](https://master-api.flutter.dev/flutter/cupertino/CupertinoContextMenu/previewBuilder.html).  If they want to control their own border radius and shadow during the animation, or even do something crazy and much more complicated, then they can do that themselves.

Part of Flutter's powerful composability comes from using plain generic widgets for everything and not treating any as special.  From the [styleguide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-interleaving-multiple-concepts-together):

  > Many Widgets take a child. Widgets should be entirely agnostic about the type of that child. Don’t use is or similar checks to act differently based on the type of the child.